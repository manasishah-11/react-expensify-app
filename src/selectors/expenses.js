import moment from 'moment';

const getVisibleExpenses = (expenses, {text, sortBy, startDate, endDate}) => {
    return expenses.filter((expense) => {
        const createdAtMoment = moment(expense.createdAt);
        const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, 'day') : true;
        const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day') : true;
        const TextMatch = expense.description.toLowerCase().includes(text.toLowerCase());
        return startDateMatch && endDateMatch && TextMatch;
    }).sort((a, b) => {
        if(sortBy === 'date'){
            return a.createdAt < b.createdAt ? 1 : -1;
        }
        else if(sortBy === 'amount'){
            return a.amount < b.amount ? 1: -1;
        }
    });
};

export default getVisibleExpenses;