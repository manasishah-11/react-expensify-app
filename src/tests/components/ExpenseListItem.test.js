import ExpenseListItem from '../../components/ExpenseListItem.js';
import {shallow} from 'enzyme';
import React from 'react';
import expenses from '../fixtures/expenses.js';

test('should render list item correctly', () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[0]} />);
    expect(wrapper).toMatchSnapshot();
});
