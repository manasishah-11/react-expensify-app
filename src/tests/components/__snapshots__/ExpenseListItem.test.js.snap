// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render list item correctly 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      Gum
    </h3>
  </Link>
  <p>
    $1.95
    -
    1st January, 1970
  </p>
</div>
`;
